### PR TITLE
Bug 1068691 - set aria role and label for close

### PR DESCRIPTION
**Pull request by jobara**
Set the aria role (button) and label for the close button on the message-list edit view.

### CODE REVIEW

**Comment by try-server-hook** [comment]
[jobara Justin Obara (jobara)](https://github.com/jobara) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=313498eda83e)

File: apps/email/js/cards/message_list.html [hunk 1]
```diff
@@ -33,8 +33,9 @@
          class="msg-listedit-header collapsed" role="region">
   <header>
     <a href="#" data-event="click:setEditModeDone"
-       class="msg-listedit-cancel-btn">
-      <span class="icon icon-close"></span>
+       class="msg-listedit-cancel-btn" role="button">
+      <span class="icon icon-close" data-l10n-id="message-listEdit-close">
```
Comment (yzen) [inline apps/email/js/cards/message_list.html]: Put a label on the anchor (button) itself instead. Also change the name to 'close-button' as it is pretty general and we might have to reuse it elsewhere.

File: apps/email/locales/email.en-US.properties [hunk 2]
```diff
@@ -673,6 +673,11 @@ message-list-top-action=Top
 # on the screen.
 message-list-menu.ariaLabel=Folders and settings
 
+# LOCALIZATION NOTE(message-listEdit-close): For the message-list edit view,
+# a label for a header icon button that closes the message-list edit view.
+# The following string is spoken by screen readers and not shown on the screen.
+message-listEdit-close.ariaLabel=Close
```
Comment (yzen) [inline apps/email/locales/email.en-US.properties]: See above, change the name to close-button and also move it to the top of the file besides back-button entry.

**Comment by jobara** [comment]
@yzen thanks for the review. I've made the changes you recommended.

**Comment by try-server-hook** [comment]
[jobara Justin Obara (jobara)](https://github.com/jobara) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=f8e7c77034ec)

**Pull request merged**